### DISCUSSION TITLE
"Real time" updates to network weights?

### DISCUSSION THREAD

**Discussion by McLemoreT** [coreylammie/MemTorch#112] [Q&A]
Good afternoon. I have been using this software for a few weeks now and I find it very interesting, and opens the door to a whole world of experiments and research. I have some ideas for things I would like to test, however, I think I may be confused regarding some of the functionality of the system. Does memtorch support updating the weights of the network in accordance to the nonidealities that have been applied in real time, or does it only update the weights once at the beginning of the simulation? Basically, do the weights stray further from their original value the longer I test/train/use the network? If memtorch does support this, could you please direct me toward where I can read about properlly implementing this? Thank you.

**Comment by coreylammie** [comment]
Hi @McLemoreT,

This is a great question! Currently, only one version of each layers weights are stored, and non-idealities are applied directly, i.e., if weights are re-programmed, and non-idealities are re-applied, effects will be magnified/worsened. 

I have created a new issue detailing how support/functionality can be added (#119), so that this does not occur. Although crude, for the time being, `copy.deepcopy()` can be used to clone/copy the current state of a network before applying non-idealities, to prevent compounding.
